Replicator/tests/ReplicatorCollectionTest.cc

```diff
@@ -848,4 +848,250 @@ TEST_CASE_METHOD(ReplicatorCollectionTest, "Replicate Encrypted Properties with
+
+        C4Collection* roses2  = getCollection(db2, Roses);
+        C4Collection* tulips2 = getCollection(db2, Tulips);
+        c4::ref<C4Document> rose1
+        = c4coll_getDoc(roses2, "db-roses-1"_sl, true, kDocGetMetadata, ERROR_INFO());
+        c4::ref<C4Document> tulip1
+        = c4coll_getDoc(tulips2, "db-tulips-1"_sl, true, kDocGetMetadata, ERROR_INFO());
+        
+        CHECK(!rose1);
+        CHECK( tulip1);
+    }
+
+    SECTION("PULL") {
+        C4ReplicatorValidationFunction pullFilter
+            = [](C4CollectionSpec collectionSpec,
+                 C4String docID,
+                 C4String revID,
+                 C4RevisionFlags,
+                 FLDict body,
+                 void* context) {
+                CHECK(collectionSpec == Tulips);
+                slice drop {(const char*)context };
+                return drop != docID;
+            };
+        _updateClientOptions = [=](const repl::Options& opts) {
+            repl::Options ret = opts;
+            for (repl::Options::CollectionOptions& o : ret.collectionOpts) {
+                if (repl::Options::collectionPathToSpec(o.collectionPath) == Tulips) {
+                    o.pullFilter = pullFilter;
+                    o.callbackContext = (void*)"db-tulips-1";
+                }
+            }
+            return ret;
+        };
+
+        // "db-tulips-1" is filtered out as it is pulled from db/Tulips to db2/Tulips
+        _expectedDocumentCount = 19;
+        // pull filter will generate errors for failed documents.
+        _expectedDocPullErrors = set<string>{"db-tulips-1"};
+        runPullReplication({Tulips, Lavenders, Roses}, {Roses, Tulips});
+
+        // db2 is the active client.
+        C4Collection* roses2  = getCollection(db2, Roses);
+        C4Collection* tulips2 = getCollection(db2, Tulips);
+        c4::ref<C4Document> rose1
+        = c4coll_getDoc(roses2, "db-roses-1"_sl, true, kDocGetMetadata, ERROR_INFO());
+        c4::ref<C4Document> tulip1
+        = c4coll_getDoc(tulips2, "db-tulips-1"_sl, true, kDocGetMetadata, ERROR_INFO());
+
+        CHECK( rose1);
+        CHECK(!tulip1);
+    }
+
+    SECTION("DocIDs on PULL") {
+        fleece::Encoder enc;
+        enc.beginArray();
+        enc.writeString("db-tulips-2"_sl);
+        enc.writeString("db-tulips-7"_sl);
+        enc.writeString("db-tulips-4"_sl);
+        enc.endArray();
+        Doc docIDs {enc.finish()};
+        _updateClientOptions = [&](const repl::Options& opts) {
+            repl::Options ret = opts;
+            for (repl::Options::CollectionOptions& o : ret.collectionOpts) {
+                if (repl::Options::collectionPathToSpec(o.collectionPath) == Tulips) {
+                    o.setProperty(slice(kC4ReplicatorOptionDocIDs), docIDs.root());
+                }
+            }
+            return ret;
+        };
+
+        // Only 3 docs from db are pulled to db2. db2 is the active replicator.
```

Comment:
Misleading comment.  Only 3 docs are pulled from TULIPS, but 13 are pulled in total.